### PR TITLE
Pot Texture Match Contents

### PR DESCRIPTION
**Pull request by SlyryD**
Pot Texture Match Contents

### CODE REVIEW

File: ASM/c/pots.c [hunk 1]
```diff
@@ -2,72 +2,66 @@
...
+extern Mtx_t *write_matrix_stack_top(z64_gfx_t *gfx);
 asm(".equ write_matrix_stack_top, 0x800AB900");
 
-void get_pot_override(z64_actor_t *actor) {
-    uint8_t size  = ((uint8_t*)actor)[0x01E9];
-    uint8_t color = size;
+override_t get_pot_override(z64_actor_t *actor, z64_game_t *game) {
+    // make a dummy EnItem00 with enough info to get the override
+    EnItem00 dummy;
+    dummy.collectibleFlag = (actor->variable & 0x3F00) >> 8;
```
Comment (rrealmuto) [inline ASM/c/pots.c]: the collectible flag for a pot is in mask 0x7E00

File: ASM/c/pots.c [hunk 2]
```diff
@@ -2,72 +2,66 @@
...
+    dummy.actor.actor_id = 0x15;
+    dummy.actor.dropFlag = 1;
+    dummy.actor.dropFlag |= (actor->variable & 0x001F) >> 5;
```
Comment (rrealmuto) [inline ASM/c/pots.c]: don't need this (it actually won't do anything as currently written). I add 2 additional bits of flag space to mask 0xC0 of the actor variable for en_item00s when dropped from things like crates. Pots don't currently use this.

File: ASM/c/pots.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+void draw_pot(z64_actor_t *actor, z64_game_t *game) {
+    z64_gfx_t *gfx = game->common.gfx;
+    Gfx *opa_ptr = gfx->poly_opa.p;
+
+    //write matrix
+    Mtx_t *mtx = write_matrix_stack_top(gfx);
+
+    //get chest_type
+    override_t override = get_pot_override(actor, game);
+    uint16_t resolved_item_id = resolve_upgrades(override.value.item_id);
```
Comment (SlyryD) [inline ASM/c/pots.c]: There appears to be an exception thrown in resolve_upgrades, causing a crash.

Comment (rrealmuto) [reply ASM/c/pots.c]: looks like the only way resolve_upgrades would throw an exception would be if get_item_row returns NULL. Check what value of item_id is getting passed in.

Comment (rrealmuto) [reply ASM/c/pots.c]: also you should set dummy.actor.dropFlag = 1;

**Pull request closed without merging**